src/libcore/iter.rs

```diff
@@ -98,6 +98,13 @@ pub trait Iterator {
     ///
     /// An upper bound of `None` means either there is no known upper bound, or
     /// the upper bound does not fit within a `usize`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let a = [1, 2, 3, 4, 5];
+    /// assert_eq!((5, Some(5)), a.iter().size_hint());
```

Comment:
It is not good example at all. While in practice this particular iterator over this particular data type will always return the exact size bounds, the purpose of this method (as the name implies) is for it to act as a hint. None of that hint-ness is at all reflected in this example.

Perhaps various iterator adaptors would make the hint more interesting and fitting? For example:

```
let iterator = (0..10).filter(|x| x % 2 == 0).chain(15..20);
iterator.size_hint() // returns (5, Some(15)) and the actual length of 10 is indeed in-bounds.
```